src/probnum/randvars/_random_variable.py

```diff
@@ -446,7 +418,9 @@ def in_support(self, x: _ValueType) -> bool:
 
         return in_support
 
-    def sample(self, size: ShapeArgType = ()) -> _ValueType:
+    def sample(
+        self, size: ShapeArgType = (), random_state: RandomStateArgType = None
```

Comment:
I don't think it is a good idea to allow passing seeds directly to the sample method, since in this case a random state object is created for one sample process only. IMHO this is weird behavior and might also lead to weird behavior, since a user gets the exact same samples when calling sample twice with a seed as `random_state`, e.g. in a loop or in some deeply nested code.

I think we should only allow `random_state` args of type `RandomStateType`, i.e. only `np.random.Generator`s and `np.random.RandomState`s. If `None` is given, a global `np.random.Generator` should be used. At a future point, we might want to drop support for the deprecated `np.random.RandomState` alltogether.